packages/common/src/api/tan-query/coins/useCoinLeaderboard.ts

```diff
@@ -0,0 +1,107 @@
+import { decodeHashId } from '@audius/sdk'
+import { InfiniteData, useInfiniteQuery } from '@tanstack/react-query'
+
+import { useQueryContext } from '~/api/tan-query/utils'
+import { TOKEN_LISTING_MAP } from '~/store'
+
+import { QUERY_KEYS } from '../queryKeys'
+import { QueryKey, QueryOptions } from '../types'
+
+const DEFAULT_PAGE_SIZE = 20
+
+export interface CoinMember {
+  user_id: number
+  balance: number
+}
+
+export interface UseCoinLeaderboardArgs {
+  mint: string | null | undefined
+  pageSize?: number
+  minBalance?: number
+  sortDirection?: 'asc' | 'desc'
+}
+
+export const getCoinLeaderboardQueryKey = (
+  mint: string | null | undefined,
+  pageSize: number = DEFAULT_PAGE_SIZE,
+  minBalance?: number,
+  sortDirection?: 'asc' | 'desc'
+) =>
+  [
+    QUERY_KEYS.coinLeaderboard,
+    mint,
+    pageSize,
+    minBalance,
+    sortDirection
+  ] as unknown as QueryKey<InfiniteData<CoinMember[]>>
+
+export const useCoinLeaderboard = (
```

Comment:
maybe let's call this `useArtistCoinMembers` since it's not necessarily only for the leaderboard?